### PR TITLE
#530 [Added] Machine Learning Roadmap

### PR DESCRIPTION
**Pull request by harsh-singh-repository**
I added the machine learning roadmap with python that includes more practical way to learn machine learning including the specialized topics ,podcast and research.

## Fixes Issue
- Added the Machine Learning Roadmap

## Changes proposed
- Add the Machine learning roadmap with the best resources
- Includes practical approach for machine Learning
- Also includes Research papers and other advanced resources as well

### CODE REVIEW

File: Machine-Learning/README.md [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+
+## Getting Started with Python
+
+- [The Python tutorial -python.org](https://docs.python.org/3/tutorial/)
+- [Python in One Short -FreeCodeCamp(Youtube)](https://youtu.be/eWRfhZUzrAch)
+- [Python for absolute Begineer  -Corey Schafer(Full playlist including frameworks  and libraries)](https://youtube.com/playlist?list=PL-osiE80TeTt2d9bfVyTiXJA-UTHn6WwU)
+- [Intermidiate python in 6 hours -FreeCodeCamp](https://youtu.be/HGOBQPFzWKo)
+- [Python Tutorial -Tutorials point](https://www.tutorialspoint.com/python/index.html)
+- [Data Structures and algorithms in python -FreeCodeCamp](https://youtu.be/pkYVOmU3MgA)
+-[Robust Python: Write Clean and Maintainable Code by patrick viafore-pdf](https://dokumen.pub/download/robust-python-write-clean-and-maintainable-code-1nbsped-1098100662-9781098100667-c-7156903.html)
```
Comment (inclinedadarsh) [inline Machine-Learning/README.md:16]: This is a pirated PDf of the book.

Comment (inclinedadarsh) [reply Machine-Learning/README.md:16]: Remove this.

File: Machine-Learning/README.md [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+
+- [What is Machine Learning? -IBM.com](https://www.ibm.com/topics/machine-learning)
+- [What is Machine Learning? -MIT](https://mitsloan.mit.edu/ideas-made-to-matter/machine-learning-explained)
+
+## Getting Started with Python
+
+- [The Python tutorial -python.org](https://docs.python.org/3/tutorial/)
+- [Python in One Short -FreeCodeCamp(Youtube)](https://youtu.be/eWRfhZUzrAch)
+- [Python for absolute Begineer  -Corey Schafer(Full playlist including frameworks  and libraries)](https://youtube.com/playlist?list=PL-osiE80TeTt2d9bfVyTiXJA-UTHn6WwU)
+- [Intermidiate python in 6 hours -FreeCodeCamp](https://youtu.be/HGOBQPFzWKo)
```
Comment (inclinedadarsh) [inline Machine-Learning/README.md:13]: Wrong spelling of "Intermediate"

File: Machine-Learning/README.md [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+- [Python Tutorial -Tutorials point](https://www.tutorialspoint.com/python/index.html)
```
Comment (inclinedadarsh) [inline Machine-Learning/README.md:14]: This resource doesn't exist.
Link leads to a 404 page.

File: Machine-Learning/README.md [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+- [Data Structures and algorithms in python -FreeCodeCamp](https://youtu.be/pkYVOmU3MgA)
+-[Robust Python: Write Clean and Maintainable Code by patrick viafore-pdf](https://dokumen.pub/download/robust-python-write-clean-and-maintainable-code-1nbsped-1098100662-9781098100667-c-7156903.html)
+
+## Get Familiar with Mathematics and Statistics
```
Comment (inclinedadarsh) [inline Machine-Learning/README.md:18]: This section should be moved to the bottom as it's not necessary for someone to get started with Machine Learning. Things such as Numpy, Pandas provide more hands on experience with Machine Learning.

File: Machine-Learning/README.md [hunk 5]
```diff
@@ -0,0 +1,106 @@
...
+
+## Exploring Data Analysis and Visualization
+- [Data analysis with Python -FreeCodeCamp](https://youtu.be/GPVsHOlRBBI)
+- [Python Numpy Tutorial -FreeCodeCamp](https://youtu.be/QUT1VHiLmmI)
+- [Metaplotlib crash course -FreeCodeCamp](https://youtu.be/3Xc3CA655Y4)
+- [Python Pandas Tutorial -Corey Schafer](https://youtube.com/playlist?list=PL-osiE80TeTsWmV9i9c58mdDCSskIFdDS)
+- [Python Metaplotlib Tutorial -Corey Schafer](https://youtube.com/playlist?list=PL-osiE80TeTvipOqomVEeZ1HRrcEvtZB_)
+- [Pandas Documnetation](https://pandas.pydata.org/docs/)
+- [Numpy Documentation](https://numpy.org/doc/)
+- [Data analysis with Python -Tutorialspoint](https://www.tutorialspoint.com/data-analysis-and-visualization-in-python)
```
Comment (inclinedadarsh) [inline Machine-Learning/README.md:45]: Many errors in the resource. Beginners will have a hard time figuring them out.

File: Machine-Learning/README.md [hunk 6]
```diff
@@ -0,0 +1,106 @@
...
+- [Data analysis with Python -GeeksforGeeks](https://www.geeksforgeeks.org/data-analysis-visualization-python/)
```
Comment (inclinedadarsh) [inline Machine-Learning/README.md:46]: Already many similar resources listed.

File: Machine-Learning/README.md [hunk 7]
```diff
@@ -0,0 +1,106 @@
...
+
+## Dive into Machine Learning
+- [Machine Learning -by Andrew Ng](https://youtube.com/playlist?list=PLoROMvodv4rMiGQp3WXShtMGgzqpfVfbU)
+- [Machine Learning -by Andrew Ng(All in One)](https://youtube.com/playlist?list=PLLssT5z_DsK-h9vYZkQkYNWcItqhlRJLN)
+- [Machine Learning full course for begineer - FreeCodeCamp](https://youtu.be/NWONeJKn6kc)
+- [Machine Learning with Python -Sentdex(Youtube)](https://youtube.com/playlist?list=PLQVvvaa0QuDfKTOs3Keq_kaG2P55YRn5v)
+- [Hands-On Machine Learning with Scikit-Learn, Keras, and TensorFlow by Aurélien Géron -pdf](http://powerunit-ju.com/wp-content/uploads/2021/04/Aurelien-Geron-Hands-On-Machine-Learning-with-Scikit-Learn-Keras-and-Tensorflow_-Concepts-Tools-and-Techniques-to-Build-Intelligent-Systems-OReilly-Media-2019.pdf)
```
Comment (inclinedadarsh) [inline Machine-Learning/README.md:53]: Pirated book.

File: Machine-Learning/README.md [hunk 8]
```diff
@@ -0,0 +1,106 @@
...
+- [Python Metaplotlib Tutorial -Corey Schafer](https://youtube.com/playlist?list=PL-osiE80TeTvipOqomVEeZ1HRrcEvtZB_)
+- [Pandas Documnetation](https://pandas.pydata.org/docs/)
+- [Numpy Documentation](https://numpy.org/doc/)
+- [Data analysis with Python -Tutorialspoint](https://www.tutorialspoint.com/data-analysis-and-visualization-in-python)
+- [Data analysis with Python -GeeksforGeeks](https://www.geeksforgeeks.org/data-analysis-visualization-python/)
+
+## Dive into Machine Learning
+- [Machine Learning -by Andrew Ng](https://youtube.com/playlist?list=PLoROMvodv4rMiGQp3WXShtMGgzqpfVfbU)
+- [Machine Learning -by Andrew Ng(All in One)](https://youtube.com/playlist?list=PLLssT5z_DsK-h9vYZkQkYNWcItqhlRJLN)
+- [Machine Learning full course for begineer - FreeCodeCamp](https://youtu.be/NWONeJKn6kc)
```
Comment (inclinedadarsh) [inline Machine-Learning/README.md:51]: Pretty complex.
https://youtu.be/i_LwzRVP7bg this one is more beginner friendly.

File: Machine-Learning/README.md [hunk 9]
```diff
@@ -0,0 +1,106 @@
...
+- [Deep Learning Crash Course -FreeCodeCamp](https://youtu.be/VyWAvY2CF9c)
+- [Keras with TensorFlow Course - Python Deep Learning and Neural Networks for Beginners Tutorial -FreeCodeCamp](https://youtu.be/qFJeN9V1ZsI)
+- [Neural Networks from Scratch in Python -Sentdex](https://youtube.com/playlist?list=PLQVvvaa0QuDcjD5BAw2DxE6OF2tius3V3)
+- [Practicle Deep Learning by Fast.ai -FreeCodeCamp](https://youtu.be/0oyCUWLL_fU)
+- [Neural Network Full Course](https://youtu.be/ob1yS9g-Zcs)
+- [ Convolutional Neural Networks for Visual Recognition -Standford(Youtube)](https://youtube.com/playlist?list=PL3FW7Lu3i5JvHM8ljYj-zLfQRF3EO8sYv)
+- [Deep Learning -by Standford online](https://youtube.com/playlist?list=PLoROMvodv4rOABXSygHTsbvUz4G_YQhOb)
+- [Deep Learning with PyTorch (9-Day Mini-Course)](https://machinelearningmastery.com/deep-learning-with-pytorch-9-day-mini-course/)
+
+## Spacelization in Machine Learning
```
Comment (inclinedadarsh) [inline Machine-Learning/README.md:67]: Specialization*

File: Machine-Learning/README.md [hunk 10]
```diff
@@ -0,0 +1,106 @@
...
+### Statistics and Probablity
+- [What is Statistics (and why is it important in machine learning)?](https://machinelearningmastery.com/what-is-statistics/)
+- [Statistics and Probablity -Edureka](https://youtu.be/XcLO4f1i4Yo)
+- [Statistics by FreeCodeCamp(Youtube)](https://youtu.be/xxpc-HPKN28)
+- [Probablity - Harvard University(Youtube)](https://youtube.com/playlist?list=PL2SOU6wwxB0uwwH80KTQ6ht66KWxbzTIo)
+- [Probablity-Khan Academy(Youtube)](https://youtube.com/playlist?list=PLC58778F28211FA19)
+- [Probablity for Machine Learning - Tyler Watkins](https://machinelearningmastery.com/probability-for-machine-learning/)
+- [Best Books for studying probablity for Machine Learning](https://www.javatpoint.com/probability-and-statistics-books-for-machine-learning)
+
+## Exploring Data Analysis and Visualization
```
Comment (inclinedadarsh) [inline Machine-Learning/README.md:37]: Other good resources that should be listed:

- For Numpy Practice: https://github.com/rougier/numpy-100
- For Data Analysis: https://www.freecodecamp.org/learn/data-analysis-with-python/

File: Machine-Learning/README.md [hunk 11]
```diff
@@ -0,0 +1,106 @@
...
+## Spacelization in Machine Learning
+
+### Natural Language Processing
+- [Natural Language Processing with TensorFlow2 -by Freecodecamp](https://youtu.be/B2q5cRJvqI8)
+- [Natural Language Processing with Spacy and Python -By FreeCodeCamp](https://youtu.be/dIUTsFT2MeQ)
+- [Natural Language Processing with Python -Sentdex](https://youtube.com/playlist?list=PLQVvvaa0QuDf2JswnfiGkliBInZnIC4HL)
+- [Distributed Representations of Words and Phrases and their Compositionality" by Tomas Mikolov- Research Paper](https://proceedings.neurips.cc/paper/2013/file/9aa42b31882ec039965f3c4923ce901b-Paper.pdf)
+- [Natural Language Processing with Python by Steven Bird, Ewan Klein, and Edward Loper -Book pdf](https://tjzhifei.github.io/resources/NLTK.pdf)
+
+### Computer Vision and Image Processing
```
Comment (inclinedadarsh) [inline Machine-Learning/README.md:76]: Other good resources that should be listed:

OpenCV: https://pyimagesearch.com/
Computer Vision (OpenCV) Information: https://pyimagesearch.com/2018/07/19/opencv-tutorial-a-guide-to-learn-opencv/

File: Machine-Learning/README.md [hunk 12]
```diff
@@ -0,0 +1,106 @@
...
+-[ImageNet Classification with Deep Convolutional Neural Networks by Alex Krizhevsk -Research Paper](https://papers.nips.cc/paper_files/paper/2012/file/c399862d3b9d6b76c8436e924a68c45b-Paper.pdf)
+
+### Reinforcement Learning
+- [Reinforcement Learning in 3 hours -by Nicholas Renotte](https://youtu.be/Mut_u40Sqz4)
+- [Reinforcement Learning -by Standford](https://youtube.com/playlist?list=PLoROMvodv4rOSOPzutgyCTapiGlY2Nd8u)
+- [Reinforcement Learning full course -by FreeCodeCamp](https://youtu.be/ELE2_Mftqoc)
+- [Reinforcement Learning: An Introduction  by Richard S. Sutton and Andrew G. Barto -book pdf](https://web.stanford.edu/class/psych209/Readings/SuttonBartoIPRLBook2ndEd.pdf)
+- [Playing Atari with Deep Reinforcement Learning by Volodymyr Mnih et al -Research Paper](https://www.cs.toronto.edu/~vmnih/docs/dqn.pdf)
+
+## Practice Project
```
Comment (inclinedadarsh) [inline Machine-Learning/README.md:89]: Other projects that are more applicable in real life:

- https://pyimagesearch.com/2016/08/10/imagenet-classification-with-python-and-keras/
- https://pyimagesearch.com/2016/10/03/bubble-sheet-multiple-choice-scanner-and-test-grader-using-omr-python-and-opencv/
- https://www.youtube.com/playlist?list=PLzMcBGfZo4-lwGZWXz5Qgta_YNX3_vLS2
- https://pyimagesearch.com/2019/02/18/breast-cancer-classification-with-keras-and-deep-learning/

**Review by inclinedadarsh — CHANGES_REQUESTED** [review]

File: Machine-Learning/README.md (hunk 1 above)
Comment (siddhant-khisty) [reply Machine-Learning/README.md:16]: Please note that we want to include only free resources in all our roadmaps. Pirated content is strictly prohibited.

**Comment by inclinedadarsh** [comment]
For advanced topics:

- https://towardsdatascience.com/filtering-data-frames-in-pandas-b570b1f834b9
- https://www.youtube.com/playlist?list=PLLvvXm0q8zUZgbAaSQ5SEtE0ivbofMfg2

File: Machine-Learning/README.md (hunk 5 above)
Comment (harsh-singh-repository) [reply Machine-Learning/README.md:45]: Please suggest me some other resources which I can add to make it more beginner-friendly

File: Machine-Learning/README.md (hunk 1 above)
Comment (harsh-singh-repository) [reply Machine-Learning/README.md:16]: I am sorry for adding the pirated copy of the book

File: Machine-Learning/README.md (hunk 3 above)
Comment (harsh-singh-repository) [reply Machine-Learning/README.md:14]: Fixed the link issue actually, it was [https://www.tutorialspoint.com/python/index.htm](url)

Comment (harsh-singh-repository) [reply Machine-Learning/README.md:14]: Fixed the link issue actually it was [https://www.tutorialspoint.com/python/index.htm](url)

Comment (harsh-singh-repository) [reply Machine-Learning/README.md:14]: Fixed the link issue

File: Machine-Learning/README.md (hunk 5 above)
Comment (harsh-singh-repository) [reply Machine-Learning/README.md:45]: Remove these resources as you say there are more similar resources already present in this document.

**Pull request closed without merging**